0.50.0: test target: AttributeError: Can't get attribute 'ConfigData' on <module 'mesonbuild.coredata' from '/usr/lib/python3.7/site-packages/mesonbuild/coredata.py'>

<pre>+ /usr/bin/ninja test -v -j4 -C x86_64-redhat-linux-gnu
ninja: Entering directory `x86_64-redhat-linux-gnu&apos;
[0/1] /usr/bin/meson test --no-rebuild --print-errorlogs
Traceback (most recent call last):
  File &quot;/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py&quot;, line 122, in run
    return options.run_func(options)
  File &quot;/usr/lib/python3.7/site-packages/mesonbuild/mtest.py&quot;, line 1002, in run
    return th.doit()
  File &quot;/usr/lib/python3.7/site-packages/mesonbuild/mtest.py&quot;, line 753, in doit
    self.run_tests(tests)
  File &quot;/usr/lib/python3.7/site-packages/mesonbuild/mtest.py&quot;, line 888, in run_tests
    self.build_data = build.load(os.getcwd())
  File &quot;/usr/lib/python3.7/site-packages/mesonbuild/build.py&quot;, line 2307, in load
    obj = pickle.load(f)
AttributeError: Can&apos;t get attribute &apos;ConfigData&apos; on &lt;module &apos;mesonbuild.coredata&apos; from &apos;/usr/lib/python3.7/site-packages/mesonbuild/coredata.py&apos;&gt;
FAILED: meson-test 
/usr/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
error: Bad exit status from /var/tmp/rpm-tmp.GduYpo (%check)
</pre>

--- a/mesonbuild/build.py
+++ b/mesonbuild/build.py
@@ -2307,8 +2307,13 @@ def load(build_dir):
             obj = pickle.load(f)
     except FileNotFoundError:
         raise MesonException(nonexisting_fail_msg)
-    except pickle.UnpicklingError:
+    except (pickle.UnpicklingError, EOFError):
         raise MesonException(load_fail_msg)
+    except AttributeError:
+        raise MesonException(
+            "Build data file {!r} references functions or classes that don't "
+            "exist. This probably means that it was generated with an old "
+            "version of meson. Try running meson {} --wipe".format(filename, build_dir))
     if not isinstance(obj, Build):
         raise MesonException(load_fail_msg)
     return obj

